xarray/core/dataset.py

```diff
@@ -3867,7 +3867,7 @@ def differentiate(self, coord, edge_order=1, datetime_unit=None):
                 variables[k] = v
         return self._replace_vars_and_dims(variables)
 
-    def integrate(self, dim, datetime_unit=None):
+    def integrate(self, coord, datetime_unit=None):
```

Comment:
Should `coord=None` have the default behavior of integrating over all dimensions? Or would that be confusing in some way?

Reply:
I personally think it would be a little confusing because the result may change depending on which coordinate is used for `integrate`, e.g. if the `DataArray` has a dimension without coordinate but another one-dimensional coordinate, it is not very clear which should be used.

It would be a little convenient for 1d arrays, but aswe disallow default argument for `diff`, I like to disallow default argument here too.